openpype/hosts/webpublisher/webserver_service/webpublish_routes.py

```diff
@@ -176,23 +176,48 @@ def __init__(self, node_type, name):
 class WebpublisherBatchPublishEndpoint(_RestApiEndpoint):
     """Triggers headless publishing of batch."""
     async def post(self, request) -> Response:
-        # for postprocessing in host, currently only PS
-        host_map = {"photoshop": [".psd", ".psb"]}
+        # Validate existence of openpype executable
+        openpype_app = self.resource.executable
+        if not openpype_app or not os.path.exists(openpype_app):
+            msg = "Non existent OpenPype executable {}".format(openpype_app)
+            raise RuntimeError(msg)
 
+        # for postprocessing in host, currently only PS
         output = {}
         log.info("WebpublisherBatchPublishEndpoint called")
         content = await request.json()
 
-        batch_path = os.path.join(self.resource.upload_dir,
-                                  content["batch"])
+        # Each filter have extensions which are checked on first task item
+        #   - first filter with extensions that are on first task is used
+        #   - filter defines command and can extend arguments dictionary
+        # This is used only if 'studio_processing' is enabled on batch
+        studio_processing_filters = [
```

Comment:
Will you be updating Settings for this too?

Reply:
I don't think this is modifiable. It's inner logic of this function which change how publishing will run.